classes/submission/reviewAssignment/Repository.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * @file classes/submission/reviewAssignment/Repository.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @brief A repository to find and manage review assignment.
+ */
+
+namespace PKP\submission\reviewAssignment;
+
+use APP\core\Application;
+use APP\core\Request;
+use APP\facades\Repo;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\notification\NotificationDAO;
+use PKP\plugins\Hook;
+use PKP\reviewForm\ReviewFormResponseDAO;
+use PKP\security\Role;
+use PKP\security\RoleDAO;
+use PKP\services\PKPSchemaService;
+use PKP\submission\ReviewFilesDAO;
+use PKP\submission\reviewRound\ReviewRoundDAO;
+use PKP\validation\ValidatorFactory;
+
+class Repository
+{
+    public DAO $dao;
+
+    /** @var string $schemaMap The name of the class to map this entity to its schema */
+    public string $schemaMap = maps\Schema::class;
+
+    protected Request $request;
+
+    /** @var PKPSchemaService<ReviewAssignment> $schemaService */
+    protected PKPSchemaService $schemaService;
+
+    public function __construct(DAO $dao, Request $request, PKPSchemaService $schemaService)
+    {
+        $this->dao = $dao;
+        $this->request = $request;
+        $this->schemaService = $schemaService;
+    }
+
+    /** @copydoc DAO::newDataObject() */
+    public function newDataObject(array $params = []): ReviewAssignment
+    {
+        $object = $this->dao->newDataObject();
+        if (!empty($params)) {
+            $object->setAllData($params);
+        }
+        return $object;
+    }
+
+    /** @copydoc DAO::get() */
+    public function get(int $id): ?ReviewAssignment
```

Comment:
Please add a second (optional) `$submisisonId` parameter; the `SchemaDAO::get()` function supports the concept of a parent ID, and this can be used to ensure that calling code verifies that the ID belongs to the expected submission. (I'll tag a couple of examples with a link to this comment in a sec.)